src/components/KTextArea/KTextArea.vue

```diff
@@ -229,71 +228,59 @@ export default {
 </script>
 
 <style lang="scss" scoped>
-@import '@/styles/mixins';
+/* Component styles */
 
-.k-input-wrapper {
-  display: grid;
-  margin-bottom: var(--kui-space-40, $kui-space-40);
-  width: fit-content;
+.k-textarea {
+  display: flex;
+  flex-direction: column;
+  width: 100%;
 
-  .k-textarea {
-    margin-top: var(--kui-space-80, $kui-space-80) !important;
-  }
-
-  textarea.k-input.form-control {
-    @include input-type-input;
+  // error styles
+  &.input-error {
+    .input-textarea {
+      @include inputError;
 
-    -webkit-appearance: none;
-    appearance: none;
-    font-family: var(--kui-font-family-text, $kui-font-family-text);
-    font-size: var(--kui-font-size-40, $kui-font-size-40) !important;
-    font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular) !important;
-    line-height: var(--kui-line-height-40, $kui-line-height-40) !important;
-    padding: 17px 0 0 22px;
-    resize: none;
+      &:hover {
+        @include inputErrorHover;
+      }
 
-    &.is-resizable {
-      min-height: 50px;
-      resize: vertical;
+      &:focus {
+        @include inputErrorFocus;
+      }
     }
 
-    &:focus::placeholder {
-      color: transparent;
-    }
-
-    @include textarea-default;
-    &:hover {
-      @include textarea-hover;
-    }
-    &:focus {
-      @include textarea-focus;
+    .help-text {
+      color: var(--kui-color-text-danger, $kui-color-text-danger);
     }
   }
 
-  .char-limit {
-    color: var(--kui-color-text, $kui-color-text) !important;
-    font-size: var(--kui-font-size-30, $kui-font-size-30) !important;
-    margin-left: var(--kui-space-auto, $kui-space-auto);
-    margin-top: var(--kui-space-40, $kui-space-40) !important;
-  }
+  .help-text {
+    @include inputHelpText;
 
-  .over-char-limit {
-    color: var(--kui-color-text-danger, $kui-color-text-danger);
+    // reset default margin from browser
+    margin: 0;
+    margin-top: var(--kui-space-40, $kui-space-40);
   }
 
-  .text-on-input {
-    label.hovered, label:hover {
-      color: var(--kui-color-text-primary, $kui-color-text-primary);
+  .input-textarea {
+    @include inputDefaults;
+
+    resize: none;
+
+    &.resizable {
+      resize: vertical;
+    }
+
+    &:hover {
+      @include inputHover;
     }
 
-     @include overlay-label-input;
-  }
+    &:focus {
+      @include inputFocus;
+    }
 
-  &.input-error {
-    textarea.k-input.form-control {
-      box-shadow: none !important;
-      outline: 1px solid #d44324 !important;
-      transition: color $tmp-animation-timing-2 ease;
+    &:disabled {
```

Comment:
question: Don't we need to account for `readonly` styles as well? 

Can you add an example in the docs and in the sandbox?